app/views.py

```diff
@@ -86,10 +86,61 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class EditProfileView(LoginRequiredMixin, UpdateView):
+class SocialLinkFormMixin(FormMixin):
+    def get_class_name(self):
+        return self.model if self.model else self.classname
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["default_social_links"] = SocialLink.objects.all()
+        context["has_social_links_form"] = True
+        context["social_links"] = (
+            []
+            if self.object is None
+            else SocialLinkAttachement.objects.filter(
+                object_id=self.object.pk,
+                content_type=ContentType.objects.get_for_model(self.get_class_name()),
+            )
+        )
+        return context
+
+    def form_valid(self, form):
+        try:
+            resp = super().form_valid(form)
+            CLASS_TYPE = ContentType.objects.get_for_model(self.get_class_name())
+            social_types = self.request.POST.getlist("social-type")
+            social_contents = self.request.POST.getlist("social-content")
+            pk = self.object.pk
+            new_socials = []
+            for ind in range(len(social_types)):
+                new_socials.append(
+                    SocialLinkAttachement(
+                        link=SocialLink.objects.get(name=social_types[ind]),
+                        content=social_contents[ind],
+                        content_type=CLASS_TYPE,
+                        object_id=pk,
+                    )
+                )
+
+            for social in SocialLinkAttachement.objects.filter(content_type=CLASS_TYPE, object_id=pk):
```

Comment:
From what I can tell, this is restricting to only 1 link per type, do we want this restriction? I was thinking of an overall limit of any, say 5, links.

Reply:
One of the reasons for this is say an organization might want to link themselves as well as their cause for example.

Reply:
There can be multiple links of the same time, that for loop iterates through the list of the current object (user/initiative)'s social link [attachements] and checks whether they are present in the form submitted list of links.

Reply:
Oh, okay that makes sense.